test/IECore/IndexedIOAlgo.py

```diff
@@ -0,0 +1,187 @@
+
+		del f
+
+		return fv
+
+	def makeManyDirectoryTestFile( self ) :
+		f = IECore.FileIndexedIO( "./test/FileIndexedIO.fio", [], IECore.IndexedIO.OpenMode.Write )
+
+		for d in range( 512 ) :
+			subdir = f.subdirectory( "sub_{0}".format( d ), IECore.IndexedIO.MissingBehaviour.CreateIfMissing )
+
+			fv = IECore.FloatVectorData()
+
+			for n in range( d ) :
+				fv.append( n % 3 )
+
+			name = "myFloatVector"
+			subdir.write( name, fv )
+
+		del f
+
+
+	def testCanCopyFile( self ) :
+		fv = self.makeTestFile()
+		src = IECore.FileIndexedIO( "./test/FileIndexedIO.fio", [], IECore.IndexedIO.OpenMode.Read )
+		dst = IECore.FileIndexedIO( "./test/FileIndexedIO2.fio", [], IECore.IndexedIO.OpenMode.Write )
+		copyStats = IECore.IndexedIOAlgo.copy( src, dst, 1 )
+
+		# 1 block of 1024 * sizeof(float) = 1024*4 = 4096 bytes = 1 << 12
+		self.assertEqual( copyStats[0], [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1] )
+		self.assertEqual( copyStats[1], [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4096] )
+
+		del src
+		del dst
+
+		src = IECore.FileIndexedIO( "./test/FileIndexedIO2.fio", [], IECore.IndexedIO.OpenMode.Read )
+
+		sub1 = src.subdirectory( "sub1" )
+
+		name = "myFloatVector"
+		gv = sub1.read( name )
+
+		self.assertEqual( len( fv ), len( gv ) )
+
+		for n in range( 0, 1000 ) :
+			self.assertEqual( fv[n], gv[n] )
+
+	def testCanReadFileStats( self ) :
+		self.makeTestFile()
+		src = IECore.FileIndexedIO( "./test/FileIndexedIO.fio", [], IECore.IndexedIO.OpenMode.Read )
+
+		copyStats = IECore.IndexedIOAlgo.copy( src, None, 1 )
+		self.assertEqual( copyStats[0], [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1] )
+		self.assertEqual( copyStats[1], [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4096] )
+
+		self.makeManyDirectoryTestFile()
+		src2 = IECore.FileIndexedIO( "./test/FileIndexedIO.fio", [], IECore.IndexedIO.OpenMode.Read )
+		copyStats1thread = IECore.IndexedIOAlgo.copy( src2, None, 1 )
+		copyStats8threads = IECore.IndexedIOAlgo.copy( src2, None, 8 )
+
+		self.assertEqual( copyStats1thread, copyStats8threads )
+
+		# 512 directories each containingn an increasing number of floats
+		# sub_0 : 0 * sizeof(4) = 1 block  = 0 bytes  [1,0]
+		#											  [0, 0] no blocks > 1 byte <= 2 bytes
+		# sub_1 : 1 * sizeof(4) = 1 block  = 4 bytes  [1,4]
+		# sub_2 : 2 * sizeof(4) = 2 blocks = 8 bytes  [1, 8]
+		# sub_3 : 3 * sizeof(4) = 3 blocks = 12 bytes
+		# sub_4 : 4 * sizeof(4) = 4 blocks = 16 bytes [2, 28]
+
+		self.assertEqual( copyStats1thread[0], [1, 0, 1, 1, 2, 4, 8, 16, 32, 64, 128, 255] )
+		self.assertEqual( copyStats1thread[1], [0, 0, 4, 8, 28, 104, 400, 1568, 6208, 24704, 98560, 391680] )
+
+	def testMultithreadedCopy( self ) :
+
+		self.makeManyDirectoryTestFile()
+		src = IECore.FileIndexedIO( "./test/FileIndexedIO.fio", [], IECore.IndexedIO.OpenMode.Read )
+		dst = IECore.FileIndexedIO( "./test/FileIndexedIO2.fio", [], IECore.IndexedIO.OpenMode.Write )
+
+		# Perhaps I should raise an exception instead of falling back to single threaded?
```

Comment:
I think the better answer here is to constrain the API so that its not possible to request the impossible.